src/queues.c

```diff
@@ -190,8 +190,13 @@ int queues_notification_insert(struct notification *n)
         if (!inserted && STR_FULL(n->stack_tag) && queues_stack_by_tag(n))
                 inserted = true;
 
-        if (!inserted && settings.stack_duplicates && queues_stack_duplicate(n))
+        if (!inserted && settings.stack_duplicates && queues_stack_duplicate(n)){
+                if(settings.sort == SORT_TYPE_UPDATE){
```

Comment:
this feels like a hack

Reply:
I don't think the `waiting` queue needs to be sorted. Does the timestamp of the duplicate notification get updated before sorting? If so this should work fine.

Reply:
I'm not sorting the waiting queue anymore